source/extensions/priv/server/syskey.c

```diff
@@ -0,0 +1,79 @@
+/*!
+* @file syskey.c
+* @brief Definitions for functions to retrieve the SYSKEY from the Registry
+*/
+#include "precomp.h"
+
+/*!
+* @brief Get individual component of the SysKey from the Registry.
+* @param lsaHandle Handle to the LSA Registry Key
+* @param subkeyName String containing the name of the Subkey to read from.
+* @param tmpSysKey Pointer to the string of the Syskey we are building
+* @returns Indication of sucess or failure.
+*/
+BOOL get_syskey_component(HKEY lsaHandle, char subkeyName[255], unsigned char *tmpSysKey){
+	DWORD sizeData = 9;
+	long regStatus;
+	HKEY subkeyHandle;
+	unsigned char tmpVal[16];
+	intmax_t byteComponent = 0;
+
+	regStatus = RegOpenKeyEx(lsaHandle, subkeyName, 0, KEY_READ, &subkeyHandle);
+	if (regStatus != ERROR_SUCCESS){
+		return FALSE;
+	}
+	regStatus = RegQueryInfoKey(subkeyHandle, (LPSTR)&tmpVal, &sizeData, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL);
+	if (regStatus != ERROR_SUCCESS){
+		return FALSE;
+	}
+	byteComponent = strtoimax(tmpVal, NULL, 16);
+	strncat(tmpSysKey, (char *)&byteComponent, 4);
+	RegCloseKey(subkeyHandle);
+	return TRUE;
+}
+
+/*!
+* @brief Retrieves and assembled the SYSKEY from the Registry
+* @param sysKey Pointer to the string of the Syskey we are building
+* @returns Indication of sucess or failure.
+*/
+BOOL get_syskey(unsigned char *sysKey){
+	unsigned char tmpSysKey[17];

```

Comment:
but tmpSysKey is being initialized just below haha, sorry, I need to finish reading before writing :)